pkg/capi/edit/turtles-capi.cattle.io.capiprovider.vue

```diff
@@ -0,0 +1,176 @@
+<script lang="ts">
+import Vue from 'vue';
+import CreateEditView from '@shell/mixins/create-edit-view';
+import CruResource from '@shell/components/CruResource.vue';
+import SelectIconGrid from '@shell/components/SelectIconGrid.vue';
+import { SUB_TYPE } from '@shell/config/query-params';
+
+import { PROVIDER_TYPES } from '../types/capi';
+import ProviderConfig from './ProviderConfig.vue';
+
+interface ProviderType {
+  id: string,
+  label: string,
+  description: string,
+  icon: HTMLImageElement
+  disabled: boolean,
+}
+
+export default Vue.extend({
+  name: 'CreateProvider',
+
+  components: {
+    CruResource,
+    SelectIconGrid,
+    ProviderConfig
+  },
+
+  mixins: [CreateEditView],
+
+  props: {
+
+    value: {
+      type:    Object,
+      default: null,
+    },
+
+    /**
+     * Inherited global identifier prefix for tests
+     * Define a term based on the parent component to avoid conflicts on multiple components
+     */
+    componentTestid: {
+      type:    String,
+      default: 'capi-provider-create'
+    }
+  },
+  beforeMount() {
+    if ( this.value?.spec?.name) {
+      this.selectType(this.value.spec.name);
+    }
+  },
+
+  data() {
+    const subType = this.$route.query[SUB_TYPE] || null;
+
+    return { subType };
+  },
+
+  computed: {
+
+    templateOptions() {
+      return [];
+    },
+
+    subTypes() {
+      const out: ProviderType[] = [];
+      const getters = this.$store.getters;
+
+      PROVIDER_TYPES?.forEach((provider) => {
+        addType(provider.id, provider.disabled);
+      });
+
+      return out;
+
+      function addType(id: string, disabled = false, iconClass = undefined) {
+        const label = getters['i18n/withFallback'](`cluster.provider."${ id }"`, null, id);
+        const description = getters['i18n/withFallback'](`cluster.providerDescription."${ id }"`, null, '');
+        let icon;
+
+        try {
+          icon = require(`../assets/images/providers/${ id }.svg`);
+        } catch (e) {
+          icon = require('~shell/assets/images/generic-driver.svg');
+        }
+
+        const providerType: ProviderType = {
+          id,
+          label,
+          description,
+          icon,
+          disabled
+        };
+
+        out.push(providerType);
+      }
+    },
+    showAsForm() {
+      return true;
+    }
+  },
+
+  methods: {
+
+    clickedType(obj: ProviderType) {
+      const id = obj.id;
+
+      this.$router.applyQuery({ [SUB_TYPE]: id });
+      this.selectType(id);
+    },
+
+    selectType(type: string) {
+      this.subType = type;
+      this.$emit('set-subtype', this.$store.getters['i18n/withFallback'](`cluster.provider."${ type }"`, null, type));
+    },
+  },
+});
+</script>
+
+<template>
+  <CruResource
```

Comment:
It seems a little off to have CruResource both here and in ProviderConfig; what is it needed for in this outer component?

Reply:
It allows to cancel while selecting provider